File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/factory_method/MulOperation.java
package jv.com.tencent.cain.design_pattern.factory_method;

/**
 * @author cainjiang
 * @date 2019-02-22
 * 乘法操作类
 */
public class MulOperation extends Operation {
    @Override
    public double getResult() {
        return numA * numB;
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/simple_factoty/MulOperation.kt
package kt.com.tencent.cain.desigin_pattern.simple_factoty

/**
 * @author cainjiang
 * @date 2019-02-22
 * 乘法操作类
 */
class MulOperation : Operation() {
    override fun getResult(): Double {
        return numA * numB
    }
}